job/server/src/main/java/alluxio/job/plan/migrate/MigrateDefinition.java

```diff
@@ -169,12 +169,15 @@ private static void migrate(MigrateCommand command,
     do {
       retry = false;
       try (FileInStream in = fileSystem.openFile(new AlluxioURI(source), openFileOptions);
-           FileOutStream out = fileSystem.createFile(destinationURI, createOptions)) {
+          FileOutStream out = fileSystem.createFile(destinationURI, createOptions)) {
         try {
           IOUtils.copyLarge(in, out, new byte[8 * Constants.MB]);
         } catch (Throwable t) {
           try {
             out.cancel();
+            // this is to avoid mysterious interruption behavior on filesystem.delete
+            Thread.interrupted();
```

Comment:
This is quite unusual. Can we uncover the mystery and avoid this hopefully?

Reply:
+1

Reply:
https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html
@ggezer @jiacheliu3 I think the problem here is we are throwing CancelException with wrapped Interrupted Exception so I have to check that flag according to the doc, update the code with corresponding change from